app/Http/Backend/Controllers/ProductsController.php

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $title = 'Create new products';
+        $categories = Categories::pluck('name', 'id');
+        return view('backend.products.create', compact('title', 'categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateProductsRequest $request)
+    {
+        $data = $request->all();
+
+        if ($request->hasFile('image')) {
+            $image = $request->file('image');
+            $data['image'] = Products::upload($image);
+        }
+        $products = Products::create($data);
+        $messages = [
+            'success' => 'Create new products success.',
+            'error'   => 'Create new products failed.'
+        ];
+        if ($products) {
+            Flash::success($messages['success']);
+            return redirect('/admin/products');
+        } else {
+            Flash::error($messages['error']);
+            return redirect(('/admin/products/create'));
+        }
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id id products
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $products = Products::find($id);
+        return $products;
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param int $id id products
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $title = 'Create new products';
+        $categories = Categories::pluck('name', 'id');
+        $products = Products::find($id);
+        if ($products) {
+            return view('backend.products.edit', compact('title', 'categories', 'products'));
+        } else {
+            $messages = "Product not found.";
+            Flash::error($messages);
+            return redirect(('/admin/products'));
+        }
+
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request update
+     * @param int                      $id      id products
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateProductsRequest $request, $id)
+    {
+        $products = Products::find($id);
+        if ($products) {
+            $data = $request->except('_token', '_method');
+            if ($request->hasFile('image')) {
+                $image = $request->file('image');
+                $data['image'] = Products::upload($image);
+            } else {
+                $data['image'] = $products->image;
+            }
+            $products = Products::where('id', $id)->update($data);
```

Comment:
why don't you call `$products->update($data)`. No need to add where condition, because it already in `find` function.